syntax = "proto3";

service TodoService{
    rpc ListAllTodos(ListAllTodosRequest) returns (ListAllTodosResponse){}
    rpc AddTodo(TodoRequest) returns (Todo){}
    rpc DeleteTodo(Todo) returns (EmptyResponse){}
    rpc EditTodo(Todo) returns (Todo){}
}


message TodoRequest{
    int32 id = 1;
    int32 userId = 2;
    string title = 3;
    Status status = 4; 
}
message EmptyResponse{

}
message ListAllTodosRequest{
    int32 userId = 1;
}
message Todo{
    int32 id = 1;
    string title = 2;
    Status status = 3; 
}
message ListAllTodosResponse{
    repeated Todo todos = 1;
}

enum Status {
    UNDEFINED_STATUS = 0;
    INCOMPLETE = 1;
    COMPLETE = 2;
}

